src/contexts/UserState.tsx

```diff
@@ -74,38 +81,85 @@ const initialUserContext: {
   setUserState: React.Dispatch<UserActions>
   onLogin: OnLogin
   logout: () => void
+  refreshJWT: () => void
 } = {
   userState: initialState,
   setUserState: () => {},
   onLogin: () => {},
   logout: () => {},
+  refreshJWT: () => {},
 }
 
 const UserContext = createContext(initialUserContext)
 
 export function UserProvider({ children }: UserProviderProps) {
+  const { t } = useLanguageProvider()
   const [state, dispatch] = useReducer(reducer, initialState)
   const userState = state
   const setUserState = dispatch
   const client = useApolloClient()
+  const { push } = useRouter()
+  const { preferences } = usePrefs()
+  const { showToast, clearAllToasts } = useToast()
+
+  let refreshTokenTimer = useRef(0)
```

Comment:
I use `useRef` to get a stable, single mutable value that is unaffected by re-renders and state changes.